orc8r/tools/fab/dev_utils.py

```diff
@@ -192,6 +198,70 @@ def get_hardware_id_from_vagrant(vm_name: str) -> str:
     return str(hardware_id)
 
 
+def get_hardware_id_from_docker(
+    location_docker_compose: str =
+    '../../../lte/gateway/python/integ_tests/federated_tests/docker/',
+) -> str:
+    """
+    Get the magmad hardware ID from magma running on docker from Vagrant
+    Args:
+        location_docker_compose: location of docker compose used to run FEG
+        by default lte/gateway/python/integ_tests/federated_tests/docker
+    Returns:
+        Hardware snowflake from the VM
+    """
+    with lcd('docker'), hide('output', 'running', 'warnings'), \
+            cd(location_docker_compose):
+        hardware_id = local(
+            'docker-compose exec magmad bash -c "cat /etc/snowflake"',
+        capture=True,
+        )
+    return str(hardware_id)
+
+
+def delete_agw_certs(vm_name: str):
+    """
+    Delete certificates and gw_challenge from AGW
+
+    Args:
+        vm_name: Name of the vagrant machine to use
+    """
+    with settings(warn_only=True), hide('output', 'running', 'warnings'), \
+            cd('/var/opt/magma/certs'):
+        vagrant.setup_env_vagrant(vm_name)
+        sudo('rm gateway.*')
+        sudo('rm gw_challenge.key')
+
+
+def delete_feg_certs(location_docker_compose: str):
+    """
+    Get the magmad hardware ID from magma running on docker from Vagrant
+
+    Args:
+        location_docker_compose: location of docker compose used to run FEG
+        by default lte/gateway/python/integ_tests/federated_tests/docker
+    """
+    print("delete_feg_certs is running on directory %s" % os.getcwd())
+
+    subprocess.check_call(
+        [
+            'docker-compose exec magmad bash -c '
+            '"rm -f /var/opt/magma/certs/gateway.*"',
```

Comment:
This looks like it deletes *all* certs, not just for feg. Correct me if I'm wrong

Reply:
Ah okay. Based on your previous comment, I see that we're using this for utils across magma, not just for orc8r. In which case, I think we can coalesce these two delete functions and note that it's for gws

Reply:
Note a gateway has just one set of certificates. 
So in that folder we have `root.CA` that we need to keep. The rest (gateway.crt gateway.key gw_challenge.key) can be deleted

Reply:
Yes I see. THe dev_utils was unknown to me when I started this PR

Let me check if I can move them around where they may make more sense

Reply:
My initial idea was to have a generic function. But that didnt work

Accessing to a vagrant vm and docker container with ssh is not trivial. Vagrant changes the ssh port every time it is run (depending if you have other vagrant vm running). Accessing a docker container through ssh means you have to have ssh enabled on the container. 

Best option is to keep it simple and make a function for each case.

Reply:
After checking the file I saw why I added those functions ther. dev_utils already include `get_hardware_id_from_vagrant` which is tight to vagrant. That function is used by `register_generic_gateway` which is not accurate since the function requires a vagrant vm.

Maybe `register_generic_gateway` can be refactored in future PRs

Reply:
In fact after having reviewed `dev_utils` and seeing we already have a dependence on vagrant with `get_hardware_id_from_vagrant`, it make sense to keep those functions in there. 

I will rename all those functions to include the provider and `gw` so they are not confused with orc8r functions. This way we will be able to reuse them the day we have containerized AGW
```
get_gw_hardware_id_from_vagrant
get_gw_hardware_id_from_docker
delete_gw_certs_from_vagrant 
delete_gw_certs_from_docker
```

In the future I may create a functions for
```
register_generic_gateway_from_vagrant
register_generic_gateway_from_docker

```